checks/diskspace/diskspace.go

```diff
@@ -0,0 +1,44 @@
+package diskspace
+
+import (
+	"context"
+	"fmt"
+	"golang.org/x/sys/unix"
+	"os"
+)
+
+// New creates a new Diskspace health check function
+func New(thresholdBytes uint64, directory string) func(ctx context.Context) error {
+	return func(ctx context.Context) error {
+		// Query the OS about filesystem
+		var stat unix.Statfs_t
+		err := unix.Statfs(directory, &stat)
+		if err != nil {
+			return err
+		}
+		// Get information from OS query
+		blockSize := uint64(stat.Bsize)
+		total := stat.Blocks * blockSize
+		available := stat.Bavail * blockSize
+		// Evaluate filessystem state and return any errors
+		return check(ctx, thresholdBytes, total, available)
+	}
+}
+
+// Wrapper function around New() to provide default working directory
+func NewWorkingDirectory(thresholdBytes uint64) func(ctx context.Context) error {
+	wd, err := os.Getwd()
+	if err != nil {
+		panic("Unable to get current working directory, cannot return new check")
+	}
+	return New(thresholdBytes, wd)
+}
```

Comment:
Do you think we need this in the library? I guess it could be

Reply:
Not particularly, now that I see it again. I will remove it.